src/sql/workbench/contrib/query/browser/gridPanel.ts

```diff
@@ -388,7 +390,7 @@ export abstract class GridTableBase<T> extends Disposable implements IView {
 		super();
 
 		this.options = { ...defaultGridTableOptions, ...options };
-
+		this.copyHeaders = this.configurationService.getValue<boolean>('queryEditor.results.copyIncludeHeaders');
```

Comment:
You should query this value in the copy callbacks to ensure that you have the correct value - it currently looks like if someone runs a query and then changes the value of this setting the grid will still have the original value if they then copy the results.

Reply:
Changed the implementation to check the setting during copying, and verified the setting is respected in real time.